gulpfile.js

```diff
@@ -260,15 +260,24 @@ gulp.task('lint', (done) => {
   runseq('lint-eslint', 'lint-closure', done);
 });
 
-gulp.task('generate-externs', ['clean'], () => {
+gulp.task('generate-externs', ['clean'], async () => {
   let genClosure = require('@polymer/gen-closure-declarations').generateDeclarations;
-  return genClosure().then((declarations) => {
-    fs.writeFileSync('externs/closure-types.js', `${header}${declarations}`);
-  });
+  const declarations = await genClosure();
+  await fs.writeFile('externs/closure-types.js', `${header}${declarations}`);
+});
+
+gulp.task('generate-typescript', async () => {
+  let genTs = require('@polymer/gen-typescript-declarations').generateDeclarations;
+  await del(['types/**/*.d.ts', '!types/extra-types.d.ts']);
+  const config = await fs.readJson('gen-tsd.json');
+  const files = await genTs('.', config);
+  for (const [filePath, contents] of files) {
+    await fs.outputFile(path.join('types', filePath), contents);
+  }
 });
 
 gulp.task('update-version', () => {
   gulp.src('lib/utils/boot.html')
```

Comment:
fyi, this task looks like it's missing a return statement. It should return the stream pipeline.

Reply:
Wasn't changed in this PR, but added a return.